react/src/components/InviteFolderSettingModal.tsx

```diff
@@ -126,19 +183,67 @@ const InviteFolderPermissionSettingModal: React.FC<
       onOk={handleOk}
       confirmLoading={updatePermissions.isPending}
       okText={t('button.Save')}
+      style={{ minWidth: 550 }}
+      destroyOnClose
     >
-      {_.isEmpty(shared) ? (
-        <Empty
-          description={t('data.explorer.NoSharedFolders')}
-          image={Empty.PRESENTED_IMAGE_SIMPLE}
-        />
-      ) : (
-        <Form
-          ref={formRef}
-          initialValues={_.fromPairs(
+      <Form
+        ref={formRef}
+        preserve={false}
+        initialValues={{
+          permission: 'ro',
+          ..._.fromPairs(
             _.map(shared, (item) => [`perm-${item.shared_to.uuid}`, item.perm]),
-          )}
+          ),
+        }}
+      >
+        <Card
+          size="small"
+          type="inner"
+          title={t('data.folders.ShareFolder')}
+          style={{ marginBottom: token.margin }}
         >
+          <Flex justify="stretch" align="center">
+            {/* TODO: support multi invitations */}
+            <Form.Item
+              name="emails"
+              label={t('general.E-Mail')}
+              rules={[
+                {
+                  type: 'email',
+                  message: t('data.InvalidEmail'),
+                },
+                {
+                  type: 'string',
+                  max: 64,
+                  message: t('maxLength.64chars'),
+                },
+              ]}
+              style={{ marginBottom: 0 }}
+            >
+              <Input placeholder={t('data.explorer.EnterEmailAddress')} />
+            </Form.Item>
+            <Form.Item
+              name="permission"
+              label={t('data.Permission')}
+              style={{ marginBottom: 0 }}
+            >
+              <Select
+                options={[
+                  { label: t('data.ReadOnly'), value: 'ro' },
+                  { label: t('data.ReadWrite'), value: 'rw' },
+                ]}
+                popupMatchSelectWidth={false}
+              />
+            </Form.Item>
+          </Flex>
+        </Card>
+        {_.isEmpty(shared) ? (
+          <Empty
+            description={t('data.explorer.NoSharedFolders')}
```

Comment:
The text seems out of context because it's a list of shared users, not a list of shared folders.